### PR TITLE
Add add_texture functionality to graphics lib - Fix #105

### LINKED ISSUE #105: Textures
Replace colors by textures

### PR DESCRIPTION
**Pull request by Abergard**
- additionally make white color a default color

---
This change is [Reviewable](https://reviewable.io/reviews/nadzwyczajnagruparobocza/bomberman/124)

### CODE REVIEW

File: src/graphics/includes/graphics/renderer_pool.hpp [hunk 1]
```diff
@@ -13,7 +13,7 @@ class renderer_pool
     virtual ~renderer_pool() noexcept = default;
 
     virtual renderer_id acquire(const math::Size2f&, const math::Position2f&,
-                                const color&) = 0;
+                                const color& = graphics::colors::white) = 0;
```
Comment (kn65op) [inline src/graphics/includes/graphics/renderer_pool.hpp]: Can't we override this method with option to set texture_path instead of color?

**Comment by kn65op** [comment]
I think better name for images would be resources.

**Review by kn65op — DISMISSED** [review]
:lgtm:

> Reviewed 2 of 16 files at r1, 29 of 29 files at r2.
> *[Reviewable](https://reviewable.io/reviews/nadzwyczajnagruparobocza/bomberman/124#-:-LmOV_LZ3dfsDwtmLYvI:bnfp4nl)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @moose-x86)

**Review by kn65op — CHANGES_REQUESTED** [review]
> Reviewed 8 of 8 files at r3.
> *[Reviewable](https://reviewable.io/reviews/nadzwyczajnagruparobocza/bomberman/124)* status: all files reviewed, 1 unresolved discussion (waiting on @Abergard and @moose-x86)

___
*[src/graphics/src/texture_warehous.hpp, line 12 at r3](https://reviewable.io/reviews/nadzwyczajnagruparobocza/bomberman/124#-LmTjtTY3dpYs0XJpUKF:-LmTjtTY3dpYs0XJpUKG:btf4w5c) ([raw file](https://github.com/nadzwyczajnagruparobocza/bomberman/blob/a0394a938d539626a32530a12c439f1e564d4b15/src/graphics/src/texture_warehous.hpp#L12)):*
> ```cpp
> using texture_path = std::string;
> 
> class texture_warehous
> ```

Shouldn't be werehous**e**?

**Review by Abergard — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/nadzwyczajnagruparobocza/bomberman/124)* status: all files reviewed, 1 unresolved discussion (waiting on @kn65op and @moose-x86)

___
*[src/graphics/src/texture_warehous.hpp, line 12 at r3](https://reviewable.io/reviews/nadzwyczajnagruparobocza/bomberman/124#-LmTjtTY3dpYs0XJpUKF:-LmTnHO09qSuSasFreuN:b-w13az3) ([raw file](https://github.com/nadzwyczajnagruparobocza/bomberman/blob/a0394a938d539626a32530a12c439f1e564d4b15/src/graphics/src/texture_warehous.hpp#L12)):*
<details><summary><i>Previously, kn65op (Tomasz Drzewiecki) wrote…</i></summary><blockquote>

Shouldn't be werehous**e**?
</blockquote></details>

Yes, it should be

**Review by Abergard — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/nadzwyczajnagruparobocza/bomberman/124)* status: 21 of 31 files reviewed, 1 unresolved discussion (waiting on @kn65op and @moose-x86)

___
*[src/graphics/src/texture_warehous.hpp, line 12 at r3](https://reviewable.io/reviews/nadzwyczajnagruparobocza/bomberman/124#-LmTjtTY3dpYs0XJpUKF:-LmTndlj-avOyuhDAd_9:b-896fix) ([raw file](https://github.com/nadzwyczajnagruparobocza/bomberman/blob/a0394a938d539626a32530a12c439f1e564d4b15/src/graphics/src/texture_warehous.hpp#L12)):*
<details><summary><i>Previously, Abergard (Lukass) wrote…</i></summary><blockquote>

Yes, it should be
</blockquote></details>

Done.

**Review by kn65op — APPROVED** [review]
> Reviewed 10 of 10 files at r4.
> *[Reviewable](https://reviewable.io/reviews/nadzwyczajnagruparobocza/bomberman/124)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @moose-x86)

**Pull request merged**